rs/src/parse/relations/join.rs

```diff
@@ -106,59 +105,73 @@ pub fn parse_join_rel(x: &substrait::JoinRel, y: &mut context::Context) -> diagn
-            "input for each pair where the join expression yields true. ",
-            "If the join expression never yields true for any left or ",
-            "right row, this returns a row anyway, with the fields ",
-            "corresponding to the other input set to null.",
-        ),
-        JoinType::Left => concat!(
-            " Returns rows combining the row from the left and right ",
-            "input for each pair where the join expression yields true. ",
-            "If the join expression never yields true for a row from the ",
-            "left, this returns a row anyway, with the fields corresponding ",
-            "to the right input set to null.",
-        ),
-        JoinType::Right => concat!(
-            " Returns rows combining the row from the left and right ",
-            "input for each pair where the join expression yields true. ",
-            "If the join expression never yields true for a row from the ",
-            "right, this returns a row anyway, with the fields corresponding ",
-            "to the left input set to null.",
-        ),
-        JoinType::Semi => concat!(
-            " Filters rows from the left input, propagating a row only if ",
-            "the join expression yields true for that row combined with ",
-            "any row from the right input.",
-        ),
-        JoinType::Anti => concat!(
-            " Filters rows from the left input, propagating a row only if ",
-            "the join expression does not yield true for that row combined ",
-            "with any row from the right input.",
-        ),
-        JoinType::Single => concat!(
-            " Returns a row for each row from the left input, concatenating ",
-            "it with the row from the right input for which the join ",
-            "expression yields true. If the expression never yields true for ",
-            "a left input, the fields corresponding to the right input are ",
-            "set to null. If the expression yields true for a left row and ",
-            "multiple right rows, this may return the first pair encountered ",
-            "or throw an error."
-        ),
-    }));
-    if let Some(filter_expression) = filter_expression {
-        y.push_summary(
-            comment::Comment::new()
-                .nl()
-                .plain(format!("The result is filtered by {filter_expression:#}.")),
-        );
+    let nullable = if join_expression_node.data_type().nullable() {
+        "false or null"
+    } else {
+        "false"
+    };
+    y.push_summary(
+        comment::Comment::new().nl().plain(match join_type {
+            JoinType::Unspecified => "".to_string(),
+            JoinType::Inner => format!(
+                "Returns rows combining the row from the left and right \
+                input for each pair where the join expression yields true. \
+                Discarding rows where the join expression yields {}.",
```

Comment:
This doesn't seem very grammatically correct;
```suggestion
                "Returns rows combining the row from the left and right \
                input for each pair where the join expression yields true, \
                discarding rows where the join expression yields {}.",
```
Likewise for the other options.